modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/VariantsContext.es.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import React, {createContext, useState} from 'react';
+
+import * as DefaultVariant from './DefaultVariant.es';
+
+export const VariantsContext = createContext({
+	Components: DefaultVariant,
+	setComponents: () => {},
+});
+
+export const VariantsProvider = ({children, components}) => {
+	const [Components, setComponents] = useState(components);
+
+	return (
+		<VariantsContext.Provider value={{Components, setComponents}}>
+			{children}
+		</VariantsContext.Provider>
+	);
```

Comment:
This is very strange, we don't need to store components within a state and we don't need to change components.

Reply:
Yea, I think that I left it here for the future plan of using a hook like `useVariants` to change dynamically which Components  are We using on our variants but it's a leftover. Removing...

Reply:
✅